File: app/src/main/java/com/cleveroad/aropensource/ui/screens/main/mlkit/common/VisionImageProcessor.java
package com.cleveroad.aropensource.ui.screens.main.mlkit.common;

import com.google.firebase.ml.common.FirebaseMLException;

import java.nio.ByteBuffer;

/**
 * An inferface to process the images with different ML Kit detectors and custom image models.
 */
public interface VisionImageProcessor {

    /**
     * Processes the images with the underlying machine learning models.
     */
    void process(ByteBuffer data, FrameMetadata frameMetadata, GraphicOverlay graphicOverlay)
            throws FirebaseMLException;

    /**
     * Stops the underlying machine learning model and release resources.
     */
    void stop();
}

File: lib/src/main/java/com/cleveroad/arfacedetector/ui/screens/main/mlkit/common/VisionImageProcessor.kt
package com.cleveroad.arfacedetector.ui.screens.main.mlkit.common

import com.google.firebase.ml.common.FirebaseMLException

import java.nio.ByteBuffer

/**
 * An inferface to process the images with different ML Kit detectors and custom image models.
 */
interface VisionImageProcessor {

    /**
     * Processes the images with the underlying machine learning models.
     */
    @Throws(FirebaseMLException::class)
    fun process(data: ByteBuffer, frameMetadata: FrameMetadata, graphicOverlay: GraphicOverlay)

    /**
     * Stops the underlying machine learning model and release resources.
     */
    fun stop()
}